dvc/tree/local.py

```diff
@@ -316,6 +316,15 @@ def get_file_hash(self, path_info):
 
         return hash_info
 
+    def get_dir_hash_no_cache(self, dir_info):
+        hash_info = HashInfo(
+            self.PARAM_CHECKSUM, dict_md5(dir_info.to_list()),
+        )
+        hash_info.value += self.CHECKSUM_DIR_SUFFIX
+        hash_info.dir_info = dir_info
+        hash_info.nfiles = dir_info.nfiles
+        return hash_info
```

Comment:
Ok, so it seems to me that we are duplicating code here. There is already code responsible for `dir` hash computations, in `cache/base.py`. Duplicating it here, exposes us to possibility, that we will calculate dir hash differently than we do it when `repo` is present. I think that should not happen, we should reuse the code. I do get that in that case we don't have `repo` to call. Worth noting that we have commands (`plots`, `metrics`) that don't require repo to execute.

Reply:
Redone, please take a look